File: solutions/cpp/lasagna/1/lasagna.cpp
/* The amount in minutes that the lasagna should stay in the oven. */
int ovenTime() {
    return 40;
}

/* The remaining minutes based on the actual minutes already in the oven. */
int remainingOvenTime(int actualMinutesInOven) {
    return ovenTime() - actualMinutesInOven;
}

/* An estimate of the preparation time based on the number of layers and 
  the necessary time per layer. */
int preparationTime(int numberOfLayers) {
    return numberOfLayers * 2;
}

/* The total time spent to create and bake the lasagna so far. */
int elapsedTime(int numberOfLayers, int actualMinutesInOven) {
    return preparationTime(numberOfLayers) + actualMinutesInOven;
}

File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    static final int MINUTES_PER_LAYER = 2;

    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int actual) {
        return this.expectedMinutesInOven() - actual;
    }

    public int preparationTimeInMinutes(int layers) {
        return layers * MINUTES_PER_LAYER;
    }

    public int totalTimeInMinutes(int layers, int actual) {
        return this.preparationTimeInMinutes(layers) + actual;
    }
}
